transport.go

```diff
@@ -197,13 +204,14 @@ func (ap *addrProvider) ServerAddr(id raft.ServerID) (raft.ServerAddress, error)
 
 }
 
-// type Libp2pTransport struct {
-// 	raftTrans *raft.NetworkTransport
-// }
-
 func NewLibp2pTransport(h host.Host, timeout time.Duration) (*raft.NetworkTransport, error) {
+	return NewLibp2pTransportWithProtocol(h, timeout, RaftProtocol)
+}
+
+func NewLibp2pTransportWithProtocol(h host.Host, timeout time.Duration, protocol protocol.ID) (*raft.NetworkTransport, error) {
```

Comment:
I'd prefer to add a constructor with functional options.

Reply:
(or modify the existing one, it's technically "type" breaking but I don't think that'll be an issue in practice).